src/main/java/org/folio/rest/impl/ItemStorageAPI.java

```diff
@@ -143,22 +144,15 @@ public void deleteItemStorageItemsByItemId(
 
   private CompletableFuture<Item> setEffectiveCallNumber(Map<String, String> okapiHeaders, Context vertxContext, Item item) {
     CompletableFuture<Item> completableFuture = null;
-    EffectiveCallNumberComponents components = new EffectiveCallNumberComponents();
-    if (item.getItemLevelCallNumber() != null && !item.getItemLevelCallNumber().isEmpty()) {
-      components.setCallNumber(item.getItemLevelCallNumber());
-      item.setEffectiveCallNumberComponents(components);
+    if (shouldRetreiveHoldingsRecord(item)) {
```

Comment:
Sorry, my bad. Maybe let's rename it to `shouldNotRetreiveHoldingsRecord`. Since this branch is executed when item has all properties == no needs to retrieve holding.